File: src/main/java/com/hotela/service/ExampleService.java
package com.hotela.service;

import com.hotela.error.HotelaException;
import com.hotela.model.Example;
import com.hotela.repository.ExampleRepository;
import java.util.UUID;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class ExampleService {
    private final ExampleRepository exampleRepository;

    @Autowired
    public ExampleService(ExampleRepository exampleRepository) {
        this.exampleRepository = exampleRepository;
    }

    public UUID createExample(Example example) {
        if (example.getId() == null) {
            example.setId(UUID.randomUUID());
        }

        exampleRepository.save(example);
        return example.getId();
    }

    public Example findExampleById(UUID id) {
        return exampleRepository
                .findById(id)
                .orElseThrow(() -> new HotelaException.ExampleNotFoundException(id));
    }
}

File: src/main/kotlin/com/hotela/service/ExampleService.kt
package com.hotela.service

import com.hotela.model.Example
import com.hotela.repository.ExampleRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class ExampleService(
    private val exampleRepository: ExampleRepository,
) {
    suspend fun createExample(example: Example): UUID = exampleRepository.create(example)

    suspend fun getExampleById(id: UUID): Example = exampleRepository.getById(id)

    suspend fun findExampleById(id: UUID): Example? = exampleRepository.findById(id)
}
